### DISCUSSION TITLE
[IntelliSense] Stuck on building

### DISCUSSION THREAD

**Discussion by zamachnoi** [tailwindlabs/tailwindcss#11532] [Help]
**What version of VS Code are you using?**

v1.79.2

**What version of Tailwind CSS IntelliSense are you using?**

v0.9.11

**What version of Tailwind CSS are you using?**

v3.3.2

**What package manager are you using?**

npm

**What operating system are you using?**

WSL for Windows

**Tailwind config**

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
	content: [
		"./src/pages/**/*.{js,ts,jsx,tsx,mdx}",
		"./src/components/**/*.{js,ts,jsx,tsx,mdx}",
		"./src/app/**/*.{js,ts,jsx,tsx,mdx}",
	],
	theme: {
		extend: {
			backgroundImage: {
				"gradient-radial": "radial-gradient(var(--tw-gradient-stops))",
				"gradient-conic":
					"conic-gradient(from 180deg at 50% 50%, var(--tw-gradient-stops))",
			},
		},
	},
	plugins: [],
};

```

**VS Code settings**

```json
{
	"workbench.editorAssociations": {
		"*.ipynb": "jupyter-notebook"
	},
	"editor.suggestSelection": "first",
	"vsintellicode.modify.editor.suggestSelection": "automaticallyOverrodeDefaultValue",
	"files.exclude": {
		"**/.": true,
		"**/.classpath": true,
		"**/.factorypath": true,
		"**/.project": true,
		"**/.settings": true,
		"**/.tmp": true,
		"**/.vscode": true,
		"secrets": true
	},
	"kite.showWelcomeNotificationOnStartup": false,
	"diffEditor.ignoreTrimWhitespace": false,
	"workbench.colorTheme": "Dracula Soft",
	"javascript.updateImportsOnFileMove.enabled": "always",
	"json.maxItemsComputed": 100000,
	"git.enableSmartCommit": true,
	"editor.defaultFormatter": "esbenp.prettier-vscode",
	"editor.formatOnSave": true,
	"python.defaultInterpreterPath": "C:\\Users\\Nick\\AppData\\Local\\Programs\\Python\\Python39\\python.exe",
	"[python]": {
		"editor.wordBasedSuggestions": false,
		"editor.formatOnType": true
	},
	"files.autoSave": "onFocusChange",
	"editor.multiCursorModifier": "ctrlCmd",
	"html.format.wrapLineLength": 0,
	"prettier.tabWidth": 4,
	"prettier.useTabs": true,
	"debug.onTaskErrors": "debugAnyway",
	"git.autofetch": true,
	"[typescript]": { "editor.defaultFormatter": "esbenp.prettier-vscode" },
	"[typescriptreact]": {
		"editor.defaultFormatter": "esbenp.prettier-vscode"
	},
	"github.copilot.enable": {
		"*": true,
		"plaintext": false,
		"markdown": false,
		"scminput": false,
		"yaml": false,
		"java": false,
		"cpp": true,
		"typescriptreact": true,
		"go": true,
		"javascript": false
	},
	"github.copilot.advanced": {},
	"[java]": {
		"editor.defaultFormatter": "redhat.java"
	},
	"workbench.iconTheme": "vscode-icons",
	"git.confirmSync": false,
	"workbench.startupEditor": "none",
	"go.toolsManagement.autoUpdate": true,
	"tailwindCSS.emmetCompletions": true,
	"editor.inlineSuggest.enabled": true,
	"editor.quickSuggestions": {
		"strings": true
	},
	"css.validate": false,
	"tailwindCSS.experimental.configFile": null
}

```

**Reproduction URL**

A public GitHub repo that includes a minimal reproduction of the bug. **Please do not link to your actual project**, what we need instead is a _minimal_ reproduction in a fresh project without any unnecessary code. This means it doesn't matter if your real project is private/confidential, since we want a link to a separate, isolated reproduction anyways.

https://github.com/nickolasz523/tailwind-intellisense-not-loading
**Describe your issue**

Tailwind intellisense is not working, it is just stuck on building. I
```
[Global] Creating projects: [{"folder":"/home/nickolas/Projects/with-tailwindcss-app","configPath":"/home/nickolas/Projects/with-tailwindcss-app/tailwind.config.js","isUserConfigured":false,"documentSelector":[{"pattern":"/home/nickolas/Projects/with-tailwindcss-app/tailwind.config.js","priority":0},{"pattern":"/home/nickolas/Projects/with-tailwindcss-app/src/pages/**/*.{js,ts,jsx,tsx,mdx}","priority":1},{"pattern":"/home/nickolas/Projects/with-tailwindcss-app/src/components/**/*.{js,ts,jsx,tsx,mdx}","priority":1},{"pattern":"/home/nickolas/Projects/with-tailwindcss-app/src/app/**/*.{js,ts,jsx,tsx,mdx}","priority":1},{"pattern":"/home/nickolas/Projects/with-tailwindcss-app/**","priority":3}]}]
[Global] Adding watch patterns: /home/nickolas/Projects/with-tailwindcss-app/tailwind.config.js, /home/nickolas/Projects/with-tailwindcss-app
[tailwind.config.js] Initializing...
[tailwind.config.js] Loaded Tailwind CSS config file: /home/nickolas/Projects/with-tailwindcss-app/tailwind.config.js
[tailwind.config.js] Loaded postcss v8.4.24: /home/nickolas/Projects/with-tailwindcss-app/node_modules/postcss
[tailwind.config.js] Loaded tailwindcss v3.3.2: /home/nickolas/Projects/with-tailwindcss-app/node_modules/tailwindcss
[tailwind.config.js] Building...
```

**Comment by bradlc** [comment]
Hey @nickolasz523. I'm going to move this to discussions until we can identify an issue with the extension itself.

The logs look good to me. The `Building...` log is expected and doesn't indicate that the build is "stuck".

I think your GitHub repo is private so if you can make it public I can take a closer look.

The first thing to check is that you've got the [WSL extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl) installed and that Tailwind CSS IntelliSense is installed **inside WSL**. Can you confirm that this is the case?

**Reply by zamachnoi** [reply]
Hi @bradlc I do have WSL and tailwind installed inside WSL 
![image](https://github.com/tailwindlabs/tailwindcss/assets/28397411/60f62d70-ed24-497b-9ec2-2b9a8fa492f1)
![image](https://github.com/tailwindlabs/tailwindcss/assets/28397411/3ce02e49-254f-44b2-98cd-e659924fcd4c)

https://github.com/nickolasz523/tailwind

Here is a public repository. When working on windows not inside WSL, it works which is very confusing.

**Reply by zamachnoi** [reply]
hey @bradlc just wanting to follow up on this. Do you know why this may be happening?

**Reply by raihan-suryanom** [reply]
same error here.....I'm using cva + tailwind btw

**Comment by InteR1337** [comment]
@zamachnoi Hi, have you been able to resolve this issue by a chance? I suddenly started to experience the same just today...

**Comment by vitld** [comment]
Just started to get the same issue out of nowhere. It was working properly, I kept working on a feature branch, now it stopped working, being stuck on "Building...". Even if I go back to `main` branch, restart vscode it never gets going. When I hover over classnames it shows as "Loading...". Tested reinstalling extension, tested Pre-Release version. Been stuck like this for over 10 minutes now and hogging CPU power.

![Cursor_and_Extension__Tailwind_CSS_IntelliSense_—_front-end-mono_—_Viktor](https://github.com/tailwindlabs/tailwindcss/assets/11719505/7132f15d-c5b7-41eb-94d6-370c7449b588)

**Comment by InteR1337** [comment]
My original question above was related to VSCode freezing and I thought that it was due to this "Building..." from Tailwind output, but it turns out it's not related at all.

To identify the root cause, I used the bisect extensions feature in VSCode ([VSCode docs](https://code.visualstudio.com/blogs/2021/02/16/extension-bisect)) that effectively uses the binary search to disable extensions and prompt you if the issue still reproduces or not. I identified that it was ESLint extension that caused the issue (my VSCode was freezing on save). I dig deeper and found out that if I disable some `import/***` rules (https://github.com/import-js/eslint-plugin-import) it works just fine (VSCode is not freezing). So I basically disabled such rules and now it works fine which is still weird as it was working OK before.

Tailwind still shows "Building..." but seems like that's kinda steady state and Intellisense works OK for me.

Hope this helps somebody.

**Comment by vitld** [comment]
Some further investigation on my end. Issue only arises when adding
```
{
  "tailwindCSS.experimental.classRegex": [
    ["tv\\((([^()]*|\\([^()]*\\))*)\\)", "[\"'`]([^\"'`]*).*?[\"'`]"]
  ]
}
```
to VScode settings.
As taken from here: https://www.tailwind-variants.org/docs/getting-started#intellisense-setup-optional
Any ideas why? I assume maybe the regex is too broad and catches too many other patterns, and overloads the IntelliSense?

**Reply by vitld** [reply]
Found some issues regarding this. If anyone else comes across.
* https://github.com/nextui-org/tailwind-variants/issues/127 (Open)
* https://github.com/nextui-org/tailwind-variants/issues/130 (Closed)

**Reply by vitld** [reply]
Found solution to my specific problem. If anyone else runs into it.
Use the regex found in changes in this PR:
* https://github.com/nextui-org/tailwind-variants-docs/pull/34

Other related reports (Besides the ones previously mentioned):
* https://github.com/tailwindlabs/tailwindcss-intellisense/pull/897
* https://github.com/tailwindlabs/tailwindcss-intellisense/issues/973
